Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -52,8 +52,13 @@ extension SQLiteHistory: HistoryRecommendations {
             "SELECT \(siteProjection), iconID, iconURL, iconType, iconDate, iconWidth " +
             "FROM ( \(nonRecentHistory) UNION ALL \(bookmarkHighlights) ) " +
             "LEFT JOIN \(ViewHistoryIDsWithWidestFavicons) ON \(ViewHistoryIDsWithWidestFavicons).id = historyID " +
+            "WHERE url NOT IN (SELECT \(TableActivityStreamBlocklist).url FROM \(TableActivityStreamBlocklist) )" +
```

Comment:
One issue with doing the `url NOT IN` clause here is that we're not always going to return exactly 20 (or whatever limit is set to) results because we're filtering after we've selected our 20 highlights. You're probably better moving this check inside the bookmark and non-recent history queries [here](https://github.com/mozilla-mobile/firefox-ios/pull/2143/files#diff-52aa57a317ce060364e17b2124092b80R34) and [here](https://github.com/mozilla-mobile/firefox-ios/pull/2143/files#diff-52aa57a317ce060364e17b2124092b80R47).